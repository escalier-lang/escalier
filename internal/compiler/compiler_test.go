package compiler

import (
	"testing"

	"github.com/escalier-lang/escalier/internal/ast"
	"github.com/gkampitakis/go-snaps/snaps"
)

func TestVarDecls(t *testing.T) {
	source := &ast.Source{
		Path:     "input.esc",
		Contents: "val foo = 5\nvar bar = \"hello\"\n",
	}
	output := Compile(source)
	snaps.MatchSnapshot(t, output)
}

func TestFuncDecls(t *testing.T) {
	source := &ast.Source{
		Path:     "input.esc",
		Contents: "fn add(a: number, b: number) {\n  return a + b\n}\nfn sub(a: number, b: number) { return a - b }\nval sum = add(1, 2)\n",
	}
	output := Compile(source)
	snaps.MatchSnapshot(t, output)
}

func TestArrays(t *testing.T) {
	source := &ast.Source{
		Path:     "input.esc",
		Contents: "val nums = [1, 2, 3]\nval first = nums[0]\n",
	}
	output := Compile(source)
	snaps.MatchSnapshot(t, output)
}

func TestMemberAccess(t *testing.T) {
	source := &ast.Source{
		Path:     "input.esc",
		Contents: "console.log(\"hello, world\")\n",
	}
	output := Compile(source)
	snaps.MatchSnapshot(t, output)
}
