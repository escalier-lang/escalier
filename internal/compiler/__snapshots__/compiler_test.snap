
[TestVarDecls - 1]
compiler.CompilerOutput{
    ParseErrors: {
    },
    TypeErrors: nil,
    JS:         "const foo = 5;\nconst bar = \"hello\";\n//# sourceMappingURL=./index.js.map\n",
    SourceMap:  "{\"version\":3,\"file\":\"./index.js\",\"sources\":[\"../../input.esc\"],\"sourcesContent\":[\"val foo = 5\\nvar bar = \\\"hello\\\"\\n\"],\"names\":[],\"mappings\":\"MAAI,AAAA,MAAM;MACN,AAAA,MAAM\"}",
    DTS:        "declare const foo: 5;\ndeclare let bar: \"hello\";\n",
}
---

[TestFuncDecls - 1]
compiler.CompilerOutput{
    ParseErrors: {
    },
    TypeErrors: nil,
    JS:         "function add(temp1, temp2) {\n  const a = temp1;\n  const b = temp2;\n  return a + b;\n}\nfunction sub(temp3, temp4) {\n  const a = temp3;\n  const b = temp4;\n  return a - b;\n}\nconst sum = add(1, 2);\n//# sourceMappingURL=./index.js.map\n",
    SourceMap:  "{\"version\":3,\"file\":\"./index.js\",\"sources\":[\"../../input.esc\"],\"sourcesContent\":[\"fn add(a: number, b: number) {\\n  return a + b\\n}\\nfn sub(a: number, b: number) { return a - b }\\nval sum = add(1, 2)\\n\"],\"names\":[],\"mappings\":\"AAAA,aAAO,OAAW;QAAX,AAAA;QAAW,AAAA;EAChB,OAAO,AAAA,IAAI;;AAEb,aAAO,OAAW;QAAX,AAAA;QAAW,AAAA;EAAa,OAAO,AAAA,IAAI;;MACtC,AAAA,MAAM,AAAA,IAAI,GAAG\"}",
    DTS:        "declare function add(a: number, b: number): number;\ndeclare function sub(a: number, b: number): number;\ndeclare const sum: number;\n",
}
---

[TestArrays - 1]
compiler.CompilerOutput{
    ParseErrors: {
    },
    TypeErrors: nil,
    JS:         "const nums = [1, 2, 3];\nconst first = nums[0];\n//# sourceMappingURL=./index.js.map\n",
    SourceMap:  "{\"version\":3,\"file\":\"./index.js\",\"sources\":[\"../../input.esc\"],\"sourcesContent\":[\"val nums = [1, 2, 3]\\nval first = nums[0]\\n\"],\"names\":[],\"mappings\":\"MAAI,AAAA,OAAO,CAAC,GAAG,GAAG;MACd,AAAA,QAAQ,AAAA,KAAK\"}",
    DTS:        "declare const nums: [1, 2, 3];\ndeclare const first: 1;\n",
}
---

[TestMemberAccess - 1]
compiler.CompilerOutput{
    ParseErrors: {
    },
    TypeErrors: nil,
    JS:         "console.log(\"hello, world\");\n//# sourceMappingURL=./index.js.map\n",
    SourceMap:  "{\"version\":3,\"file\":\"./index.js\",\"sources\":[\"../../input.esc\"],\"sourcesContent\":[\"console.log(\\\"hello, world\\\")\\n\"],\"names\":[],\"mappings\":\"AAAA,AAAA,AAAA,QAAQ,IAAI\"}",
    DTS:        "",
}
---
